nifi-nar-bundles/nifi-azure-bundle/nifi-azure-reporting-task/src/main/java/org/apache/nifi/reporting/azure/loganalytics/AzureLogAnalyticsReportingTask.java

```diff
@@ -0,0 +1,335 @@
+        } catch (NoSuchAlgorithmException | InvalidKeyException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @OnScheduled
+    public void setup(final ConfigurationContext context) {
+       virtualMachineMetrics = JmxJvmMetrics.getInstance();
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(LOG_ANALYTICS_WORKSPACE_ID);
+        properties.add(LOG_ANALYTICS_CUSTOM_LOG_NAME);
+        properties.add(LINUX_PRIMARY_KEY);
+        properties.add(APPLICATION_ID);
+        properties.add(INSTANCE_ID);
+        properties.add(PROCESS_GROUP_IDS);
+        properties.add(JOB_NAME);
+        properties.add(SEND_JVM_METRICS);
+        properties.add(LOG_ANALYTICS_URL_ENDPOINT_FORMAT);
+        return properties;
+    }
+
+    @Override
+    public void onTrigger(final ReportingContext context){
+        final String workspaceId = context.getProperty(LOG_ANALYTICS_WORKSPACE_ID).evaluateAttributeExpressions().getValue();
+        final String linuxPrimaryKey = context.getProperty(LINUX_PRIMARY_KEY).evaluateAttributeExpressions().getValue();
+        final boolean jvmMetricsCollected = context.getProperty(SEND_JVM_METRICS).asBoolean();
+
+        final String logName = context.getProperty(LOG_ANALYTICS_CUSTOM_LOG_NAME).evaluateAttributeExpressions().getValue();
+        final String instanceId = context.getProperty(INSTANCE_ID).evaluateAttributeExpressions().getValue();
+        final String groupIds = context.getProperty(PROCESS_GROUP_IDS).evaluateAttributeExpressions().getValue();
+        final String urlEndpointFormat = context.getProperty(LOG_ANALYTICS_URL_ENDPOINT_FORMAT).evaluateAttributeExpressions().getValue();
+        try {
+            List<Metric> allMetrics = null;
+            if(groupIds == null || groupIds.isEmpty()) {
+                ProcessGroupStatus status = context.getEventAccess().getControllerStatus();
+                String processGroupName = status.getName();
+                allMetrics = collectMetrics(instanceId, status, processGroupName, jvmMetricsCollected);
+            } else {
+                allMetrics = new ArrayList<>();
+                for(String groupId: groupIds.split(",")) {
+                    groupId =groupId.trim();
+                    ProcessGroupStatus status = context.getEventAccess().getGroupStatus(groupId);
+                    String processGroupName = status.getName();
+                    allMetrics.addAll(collectMetrics(instanceId, status, processGroupName, jvmMetricsCollected));
+                }
+            }
+            HttpPost httpPost = getHttpPost(urlEndpointFormat, workspaceId, logName);
+            sendMetrics(httpPost, workspaceId, linuxPrimaryKey, allMetrics);
+        } catch (IOException | RuntimeException e) {
+            getLogger().error("Exception in outmost block", e);
```

Comment:
The message here is a bit confusing from a user standpoint. Perhaps change to something like "Failed to publish metrics to Azure". Can probably also change the `catch` clause to just `catch (Exception e) {`

Reply:
done